File: finalRest/src/main/java/ajc/formation/project/finalRest/entity/Reseau.java
package ajc.formation.alten.finalRest.entity;

import java.sql.Date;
import java.util.Set;

import org.springframework.format.annotation.DateTimeFormat;

import com.fasterxml.jackson.annotation.JsonView;

import ajc.formation.alten.finalRest.compositekey.ReseauKey;
import ajc.formation.alten.finalRest.jsonview.Views;
import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import jakarta.persistence.EmbeddedId;
import jakarta.persistence.Entity;
import jakarta.persistence.ForeignKey;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotNull;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name="RESEAU")
@Getter
@Setter
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Embeddable
public class Reseau {

	@EmbeddedId
	@NotNull
	private ReseauKey reseauKey;

	@JsonView(Views.Common.class)
	@Column(name="CODEEXTERNE")
	private Integer codeExterne;

	@JsonView(Views.Common.class)
	@Column(name="LIBELLE",length = 60)
	private String libelle;

	@JsonView(Views.Common.class)
	@Column(name="DATEDEBUT",nullable = false)
	@NotNull									
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	private Date dateDebut; 

	@JsonView(Views.Common.class)
	@DateTimeFormat(pattern = "yyyy-MM-dd")
	@Column(name="DATEFIN")
	private Date dateFin;

	@DateTimeFormat(pattern = "yyyy-MM-dd")
	@JsonView(Views.Common.class)
	@Column(name="DATEDERNIEREMODIFICATION")
	private Date dateDerniereModification;

	@JsonView(Views.Common.class)
	@Column(name="REJET")
	private Integer rejet;

	@JsonView(Views.ReseauwithTypeReseau.class)
	@OneToMany(mappedBy = "reseau")
	@NotNull
	private Set<TypeReseau>  typeReseaux;
	
	@ManyToOne
	@JoinColumn(name="IDPPDISTRIBUTION",foreignKey =@ForeignKey(name="FK_RESEAU_PPDISTRIBUTION"))
	@JsonView(Views.ReseauWithResponsable.class)
	private PPDistribution responsable;
}

File: finalAng/src/app/model/reseau.ts
import { PPDistribution } from './ppdistribution';
export class Reseau {
  public get responsable(): PPDistribution | undefined {
    return this._responsable;
  }
  public set responsable(value: PPDistribution | undefined) {
    this._responsable = value;
  }
  public get rejet(): number | undefined {
    return this._rejet;
  }
  public set rejet(value: number | undefined) {
    this._rejet = value;
  }
  public get dateDerniereModification(): Date | undefined {
    return this._dateDerniereModification;
  }
  public set dateDerniereModification(value: Date | undefined) {
    this._dateDerniereModification = value;
  }
  public get dateFin(): Date | undefined {
    return this._dateFin;
  }
  public set dateFin(value: Date | undefined) {
    this._dateFin = value;
  }
  public get dateDebut(): Date | undefined {
    return this._dateDebut;
  }
  public set dateDebut(value: Date | undefined) {
    this._dateDebut = value;
  }
  public get libelle(): string | undefined {
    return this._libelle;
  }
  public set libelle(value: string | undefined) {
    this._libelle = value;
  }
  public get codeExterne(): number | undefined {
    return this._codeExterne;
  }
  public set codeExterne(value: number | undefined) {
    this._codeExterne = value;
  }
  public get idCodeInterneReseau(): number | undefined {
    return this._idCodeInterneReseau;
  }
  public set idCodeInterneReseau(value: number | undefined) {
    this._idCodeInterneReseau = value;
  }
  public get idReseau(): number | undefined {
    return this._idReseau;
  }
  public set idReseau(value: number | undefined) {
    this._idReseau = value;
  }
  constructor(
    private _idReseau?: number,
    private _idCodeInterneReseau?: number,
    private _codeExterne?: number,
    private _libelle?: string,
    private _dateDebut?: Date,
    private _dateFin?: Date,
    private _dateDerniereModification?: Date,
    private _rejet?: number,
    private _responsable?: PPDistribution
  ) {}
}
